src/plugins/numberline/components/numberline-tile.tsx

```diff
@@ -17,30 +16,51 @@ import {
   innerPointRadius, outerPointRadius, numberlineYBound, yMidPoint, kTitleHeight, kArrowheadTop,
   kArrowheadOffset, kPointButtonRadius, tickTextTopOffsetDefault, tickTextTopOffsetMinAndMax
 } from '../numberline-tile-constants';
-import { NumberlineToolbar } from "./numberline-toolbar";
 import NumberlineArrowLeft from "../../../assets/numberline-arrow-left.svg";
 import NumberlineArrowRight from "../../../assets/numberline-arrow-right.svg";
 import { EditableNumberlineValue } from './editable-numberline-value';
+import { TileToolbar } from "../../../components/toolbar/tile-toolbar";
+import { INumberlineToolbarContext, NumberlineToolbarContext } from './numberline-toolbar-context';
+import "./numberline-toolbar-registration";
 
 import "./numberline-tile.scss";
 
+export enum ToolbarOption {
+  Selection = "selection",
+  Filled = "filled",
+  Open = "open"
+}
+
 export const NumberlineTile: React.FC<ITileProps> = observer(function NumberlineTile(props){
-  const { documentContent, model, readOnly, scale, tileElt, onRegisterTileApi, onUnregisterTileApi } = props;
+  const { model, readOnly, tileElt, onRegisterTileApi } = props;
 
   const content = model.content as NumberlineContentModelType;
   const [hoverPointId, setHoverPointId] = useState("");
   const [_selectedPointId, setSelectedPointId] = useState(""); // Just used to rerender when a point is selected
   const ui = useUIStore();
   const isTileSelected = ui.isSelectedTile(model);
 
-  //---------------- Model Manipulation Functions -------------------------------------------------
-  const deleteSelectedPoints = useCallback(() => {
-    content.deleteSelectedPoints();
-  }, [content]);
+  /* ========================== [ Determine Point is Open or Filled ]  ========================= */
+
+  const [toolbarOption, setToolbarOption] = useState<ToolbarOption>(ToolbarOption.Selection); //"selection"
```

Comment:
I don't think the toolbar needs to re-render when the toolbarOption changes. So I'd recommend changing this to:
```suggestion
  const toolbarOptionRef = useRef<ToolbarOption>(ToolbarOption.Selection); //"selection"
```
Then you can read and write the `toolbarOptionRef.current` value